lib/utils/utils.dart

```diff
@@ -18,12 +18,15 @@ import 'dart:collection';
 import 'dart:convert';
 import 'dart:io';
 import 'dart:math';
+import 'package:flutter/cupertino.dart';
 import 'package:intl/intl.dart';
 import 'package:path/path.dart' as path_package;
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:fluttertoast/fluttertoast.dart';
+import 'package:rokwire_plugin/ui/panels/web_panel.dart';
```

Comment:
@mihail-varbanov, are you ok with importing UI components in `utils.dart`? I think we wanted to avoid this, but I'm not currently sure where it would make more sense to put these functions. Let me know if you have any thoughts. Thanks!

Reply:
Hi @shurwit and @dobromirdobrev,

I am sorry for the delayed response, I did not notice this mention across the sea of GitHub notifications that I get every day.

Adding dependency of utils from plugin UI components does not look good to me. The relation should be opposite - UI components should depend on utils and not vice versa.

Moreover, I see almost no real advantage of UrlUtils.launchUrl, except maybe parsing the Uri from String and applying the null check verification. For me this is some kind of attempt for universalism that brings more harm than value. I would remove internalWebPanel as an option for UrlUtils.launchUrl, as well as the entire UrlLaunchMode definition as it would appear as 100% duplication of url_launcher.LaunchMode, but would keep UrlUtils.launchUrl(String? url) as convenient shortcut to the url_launcher.

Reply:
Hi @mihail-varbanov, the point of this PR is to use the internal web panel. Please refer back to the original issue (#217) to see why this is necessary,

Reply:
Hi @shurwit,

I do not see a clear relation between the problem that you describe in rokwire/illinois-app#2623 and the UrlUtils.launchUrl helper as it is. All that is known is that we should not load web URLs in an in-app browser on Android platforms. Whether we should load such URLs in an external browser or a web panel is a matter of design, both methods bring their benefits and drawbacks. I consider the necessity of UrlUtils.launchUrl helper only in order to avoid the following construction repeated many times:
```
LaunchMode launchMode = Platform.isAndroid ? LaunchMode.externalApplication : LaunchMode.platformDefault;
launchUrl(uri, mode: launchMode);
```
This does not require involving WebPanel in this helper. If this was up to me I would do the helper in a way like this:
```
static Future<bool?> launchUrl(String url, { url_launcher.LaunchMode? mode}) {
  Uri? uri = Uri.tryParse(url);
  mode ??= Platform.isAndroid ? url_launcher.LaunchMode.externalApplication : url_launcher.LaunchMode.inAppWebView;
  return (uri != null) ? url_launcher.launchUrl(uri, mode: mode) : Future.value(null);
}
```

Reply:
Thanks again for the feedback @mihail-varbanov, @pmarkhennessy and I are discussing this and will get back to you soon.